src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -787,74 +787,53 @@ protected virtual JwtSecurityToken ValidateSignature(string token, TokenValidati
 
             // if the kid != null and the signature fails, throw SecurityTokenSignatureKeyNotFoundException
             string kid = jwt.Header.Kid;
-            SecurityKey securityKey = null;
+            IEnumerable<SecurityKey> securityKeys = null;
 
             if (validationParameters.IssuerSigningKeyResolver != null)
             {
-                securityKey = validationParameters.IssuerSigningKeyResolver(token, jwt, kid, validationParameters);
+                securityKeys = validationParameters.IssuerSigningKeyResolver(token, jwt, kid, validationParameters);
             }
             else
             {
-                securityKey = ResolveIssuerSigningKey(token, jwt, validationParameters);
-            }
-
-            // if the security key is resolved, try just the one key
-            if (securityKey != null)
-            { 
-                try
-                {
-                    if (this.ValidateSignature(encodedBytes, signatureBytes, securityKey, mappedAlgorithm))
-                    {
-                        IdentityModelEventSource.Logger.WriteInformation(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10242, token));
-                        jwt.SigningKey = securityKey;
-                        return jwt;
-                    }
-                }
-                catch (Exception ex)
+                var securityKey = ResolveIssuerSigningKey(token, jwt, validationParameters);
```

Comment:
ResolveIssuerSigningKey needs to return 'signingKey' from the enumeration. It currently always returns validationParameters.IssuerSigningKey.